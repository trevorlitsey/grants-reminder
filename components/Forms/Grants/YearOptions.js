import moment from 'moment';
import { string, number, func } from 'prop-types';

const YearOptions = (props) => {

	const { year, selected, handleChange } = props

	const years = getYears(year);

	return (
		<select className="u-full-width two columns" name="year" value={selected} onChange={handleChange}>
			{years.map(year =>
				<option value={year} key={year} data-test="year">{year}</option>
			)}
		</select>
	)
}

YearOptions.propTypes = {
	year: number,
	selected: number.isRequired,
	handleChange: func.isRequired,
}

export default YearOptions;


// --------------------
function getYears(thisYear = Number(moment().format('YYYY'))) {
	return [
		thisYear,
		thisYear + 1,
		thisYear + 2,
		thisYear + 3,
		thisYear + 4,
	]
}