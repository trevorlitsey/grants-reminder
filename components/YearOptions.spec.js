import React from 'react';
import { shallow } from 'enzyme';

import YearOptions from './YearOptions';

describe('YearOptions', () => {

	it('should render without crashing', () => {
		const wrapper = shallowRenderYearOptions({ year: 2018 });
		expect(wrapper).toMatchSnapshot();
	})

	it('should render five years', () => {
		const wrapper = shallowRenderYearOptions();
		const years = wrapper.find('[data-test="year"]');
		expect(years.length).toEqual(5);
	})

})

function shallowRenderYearOptions(props = {}) {
	const propsToUser = {
		selected: 2018,
		handleChange: () => { },
		...props,
	}

	return shallow(<YearOptions {...propsToUser} />);

}