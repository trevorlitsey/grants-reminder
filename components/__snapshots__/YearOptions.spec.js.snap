// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YearOptions should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <YearOptions
    handleChange={[Function]}
    selected={2018}
    year={2018}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <option
          data-test="year"
          value={2018}
        >
          2018
        </option>,
        <option
          data-test="year"
          value={2019}
        >
          2019
        </option>,
        <option
          data-test="year"
          value={2020}
        >
          2020
        </option>,
        <option
          data-test="year"
          value={2021}
        >
          2021
        </option>,
        <option
          data-test="year"
          value={2022}
        >
          2022
        </option>,
      ],
      "className": "u-full-width two columns",
      "name": "year",
      "onChange": [Function],
      "value": 2018,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "2018",
        "nodeType": "host",
        "props": Object {
          "children": 2018,
          "data-test": "year",
          "value": 2018,
        },
        "ref": null,
        "rendered": 2018,
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "2019",
        "nodeType": "host",
        "props": Object {
          "children": 2019,
          "data-test": "year",
          "value": 2019,
        },
        "ref": null,
        "rendered": 2019,
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "2020",
        "nodeType": "host",
        "props": Object {
          "children": 2020,
          "data-test": "year",
          "value": 2020,
        },
        "ref": null,
        "rendered": 2020,
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "2021",
        "nodeType": "host",
        "props": Object {
          "children": 2021,
          "data-test": "year",
          "value": 2021,
        },
        "ref": null,
        "rendered": 2021,
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "2022",
        "nodeType": "host",
        "props": Object {
          "children": 2022,
          "data-test": "year",
          "value": 2022,
        },
        "ref": null,
        "rendered": 2022,
        "type": "option",
      },
    ],
    "type": "select",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            data-test="year"
            value={2018}
          >
            2018
          </option>,
          <option
            data-test="year"
            value={2019}
          >
            2019
          </option>,
          <option
            data-test="year"
            value={2020}
          >
            2020
          </option>,
          <option
            data-test="year"
            value={2021}
          >
            2021
          </option>,
          <option
            data-test="year"
            value={2022}
          >
            2022
          </option>,
        ],
        "className": "u-full-width two columns",
        "name": "year",
        "onChange": [Function],
        "value": 2018,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "2018",
          "nodeType": "host",
          "props": Object {
            "children": 2018,
            "data-test": "year",
            "value": 2018,
          },
          "ref": null,
          "rendered": 2018,
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "2019",
          "nodeType": "host",
          "props": Object {
            "children": 2019,
            "data-test": "year",
            "value": 2019,
          },
          "ref": null,
          "rendered": 2019,
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "2020",
          "nodeType": "host",
          "props": Object {
            "children": 2020,
            "data-test": "year",
            "value": 2020,
          },
          "ref": null,
          "rendered": 2020,
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "2021",
          "nodeType": "host",
          "props": Object {
            "children": 2021,
            "data-test": "year",
            "value": 2021,
          },
          "ref": null,
          "rendered": 2021,
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "2022",
          "nodeType": "host",
          "props": Object {
            "children": 2022,
            "data-test": "year",
            "value": 2022,
          },
          "ref": null,
          "rendered": 2022,
          "type": "option",
        },
      ],
      "type": "select",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
